README.md

```diff
@@ -1,3 +1,132 @@
 # repo-template
 
 **This is not an official Google product.**
+
+This template is for creating an AOD (access on demain) repository for GCP organizations, folders, and projects.
+
+After completing the installation. Follow below steps to file an AOD request.
+1. Open a PR
+  - A title and description to provide enough justification. E.g. "AOD request: debug customer issue X".
+  - Add following file at the repo root, note that any changes with below file changes will be considered AOD request and thus cannot be merged:
+    - iam.yaml - Request for IAM permissions on organization / folder / project level. To see an example of this file, refer to the example-iam.yaml file.
+  - Optionally use predefined duration labels on the PR to specify the IAM permission expiration. The duration is default to 2h if no label is used.
+
+2. PR approval
+  - Github workflows will check if the PR is an AOD request, marked by a "do_not_merge" failure workflow, and if it is a valid AOD request.
+  - Upon PR approval, the request will be handled by AOD.
+
+3. Close the PR after the request is handled successfully.
+
+## Installation
+
+1.  Create an AOD repository using this template, only copy main branch is requried.
+2.  Setup [Workload Identity Federation](https://cloud.google.com/iam/docs workload-identity-federation).
+  - Create workload identity pool and provider in a GCP project where appropriate.
+  - Create a GCP service account that identities from this pool can impersonate.
+  - Grant the service account the required IAM roles to perform AOD operations.
+    -  The minimum permissions are the get and set IAM policy. These permissions can be on projects, folders, or organizations level. For example, ["roles/resourcemanager.projectIamAdmin"](https://cloud.google.com/resource-manager/docs/access-control-proj#resourcemanager.projectIamAdmin) is a predefined role for managing project level IAM permission, an alternative is to use custom roles for a more restricted set of permissions.
+
+Below is an example of how you setup Workload Identity Federation via terraform.
+
+```terraform
```

Comment:
Can we reference some of existing module instead of dumping all the Terraform here?

Reply:
Terraform also isn't easy to understand. How about we delete these for now and follow up. I'm thinking referencing https://github.com/google-github-actions/auth#setting-up-workload-identity-federation first then reference terraform.